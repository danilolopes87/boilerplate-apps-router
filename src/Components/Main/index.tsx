import React from 'react';

const Main: React.FC = () => {
  return (
    <div>
      <h1>React AvanÃ§ado</h1>
    </div>
  );
};

export default Main;
