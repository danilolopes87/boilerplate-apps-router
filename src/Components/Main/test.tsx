import { render, screen } from '@testing-library/react';
import Main from './index'; // Ajuste o caminho conforme necessário
import React from 'react';

describe('<Main />', () => {
  it('should render heading', () => {
    //renderiza o componente
    const {container} = render(<Main />);

    //busca o elemento e verifica a existência dele
    expect(
      screen.getByRole('heading', { name: /react avançado/i})
    ).toBeInTheDocument()

    //gerar snapshot
    expect(container.firstChild).toMatchSnapshot()
  });
});
