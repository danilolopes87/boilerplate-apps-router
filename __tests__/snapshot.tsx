/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import Main from "src/Components/Main";

it("renders homepage unchanged", () => {
  const { container } = render(<Main />);
  expect(container).toMatchSnapshot();
});
