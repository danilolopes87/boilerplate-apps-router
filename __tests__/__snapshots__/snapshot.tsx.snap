// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div>
    <h1>
      React Avan√ßado
    </h1>
  </div>
</div>
`;
